static/dbstructure.config.php

```diff
@@ -1292,6 +1292,19 @@
 			"guid" => ["guid(64)"],
 		]
 	],
+	"tag" => [
+		"comment" => "item tags and mentions",
+		"fields" => [
+			"uri-id" => ["type" => "int unsigned", "not null" => "1", "primary" => "1", "relation" => ["item-uri" => "id"], "comment" => "Id of the item-uri table entry that contains the item uri"],
+			"type" => ["type" => "tinyint unsigned", "not null" => "1", "default" => "0", "primary" => "1", "comment" => ""],
+			"name" => ["type" => "varchar(64)", "not null" => "1", "default" => "", "primary" => "1", "comment" => ""],
```

Comment:
This table is bound to grow to about the same size as the one it's supposed to replace. The problem is that there are multiple instances of the same name over and over. If we want to pare down the size of the database, we need two tables: `tag (id, type, name, url)` and `tag_uri (tag_id, uri_id)`. The `name` column will be able to be longer than just 64 characters (why this limitation?) and the overall indices will be smaller as well.

Reply:
We can't use the full 255 size, since the `name` is part of the unique key. But also a tag longer than 64 chars is more an insult than helpful. I also looked into my database and found out that currently 0.25 permill of the entries in the `term` table are longer than 64 chars.

The idea with splitting it into two tables is interesting. I have to check how much we gain from it.

Reply:
What's the maximum length we can afford with regard to the unique key? That's what we should be shooting for. 1 for 4000 is still 1 truncated tag per 4000 which probably will be noticed by users.

Reply:
I guess we can use up to 160 chars. But l'm just not a big fan of #myhashtagislongerthanyourhashtagandicanmakeitevenlongerjusttoannoyyouandthewholewordbymakingitevenlongerthananythingthatdoesmakesenseatalljustbecauseican

Reply:
Me neither but it isn't always about what we are fan of, it's about what users expect.

Reply:
With having the `post` table this way, we cannot use up to 160 chars anymore when we want to respect the 767 byte length limit in InnoDB. Every `varchar` or `char` field uses 4 byte. So we can store 191 characters. Because of this limit, the `url` is `varbinary`. So we only use 255 byte (and 4 byte unicode counts as 4 characters). So we have 512 bytes left. Minus 1 byte for `type` we do have 511 bytes left - which means that we can use around 120 chars at maximum.

Reply:
Please keep #Rindfleischettiketierungsüberwachungsaufgabenübertragungsgesetz in mind. It's a real word and it's longer than 64 characters. :-)

Reply:
But that law isn't active anymore, so it doesn't count :grimacing: